pkg/node/manager/manager.go

```diff
@@ -205,6 +218,20 @@ func (m *manager) UpdateNode(nodeName string) error {
 	return nil
 }
 
+func (m *manager) GetNodeUpdateStatus(k8sNode *v1.Node, cachedNode node.Node) NodeUpdateStatus {
```

Comment:
_minor_ - Can be refactored when visiting this again. 

a) Since all of these are have return statement, else and else is not required.
b) Also, cache check could have happen first, and then isSelectedForManagement.  (For the reader, that seems a bit easier to reason about).

Reply:
Have taken up an AI to fix this in next versions.